control-plane/csi-driver/src/bin/controller/controller.rs

```diff
@@ -74,6 +78,66 @@ fn parse_protocol(proto: Option<&String>) -> Result<VolumeShareProtocol, Status>
     }
 }
 
+/// Get app node where volume frontend is situated.
+fn get_volume_app_node(volume: &Volume) -> Option<String> {
+    match volume.spec.target.as_ref() {
+        // todo: Might need to deal with multiple app nodes later?
+        Some(target) => target
+            .frontend_nodes
+            .as_ref()
+            .map(|nodes| nodes[0].name.clone()),
+        None => None,
+    }
+}
+
+async fn issue_fs_freeze(endpoint: String, volume_id: String) -> Result<(), Status> {
+    trace!(
+        "Issuing fs freeze for volume {}, endpoint {}",
+        volume_id,
+        endpoint
+    );
```

Comment:
I think you can also add enter automatically

```suggestion
#[tracing::instrument]
async fn issue_fs_freeze(endpoint: String, volume_id: String) -> Result<(), Status> {
    trace!("Issuing fs freeze");
```

Reply:
Would this work, don't think telemetry is setup.

Reply:
doh yeah this won't work, it's not that telemetry is not setup, but tracing is not configured to output fields to stdout...

Reply:
My previous comment is incorrect, tracing is configured to output fields to the output so we can do what I suggested initially